### DISCUSSION TITLE
linkWithEmailAndPassword: The Firebase ID token has been revoked

### DISCUSSION THREAD

**Discussion by sklink** [capawesome-team/capacitor-firebase#575] [Q&A] [answered]
My onboarding process starts with:
```
await FirebaseAuthentication.signInAnonymously();
```

Makes some API calls, which successfully use the anonymous ID token to authenticate.

Follows up later with:
```
await FirebaseAuthentication.linkWithEmailAndPassword({ email, password });
```

Any API calls following this fail with 401 Unauthorized and a message on the server of "The Firebase ID token has been revoked".

For every API call I'm always using: `(await FirebaseAuthentication.getIdToken()).token` to get the token. I have tried using the `{ forceRefresh: true }` option with no luck.

I should note that I'm testing this on web. I have other authentication methods working on web: currently phone, Google, and Facebook.

If any additional code would be helpful please let me know. I'm going to attempt to recreate the issue in a new repository but if anything stands out please let me know.

**Comment by robingenz** [comment]
This should be an issue with the Firebase JS SDK so I would try to find the answer there.

**Reply by sklink** [reply]
I'm migrating from the JS SDK to `capacitor-firebase/authentication`. Previously had email and password signup working with `EmailAuthProvider` and `linkWithCredential`. 

Wasn't able to find anything relevant in their GitHub Issues history. I'll try using the Firebase JS SDK in combination with `capacitor-firebase/authentication` to see if that fixes it.

**Reply by sklink** [reply]
It works if I remove:
```
await FirebaseAuthentication.linkWithEmailAndPassword({ email, password });
```
and use:
```
const credential = EmailAuthProvider.credential(email, password);
const currAuth = getAuth();
await linkWithCredential(currAuth.currentUser, credential);
await signInWithCredential(auth, credential);

// Call API using token from await FirebaseAuthentication.getIdToken()
```

Is it possible that I need to sign in after calling linkWithEmailAndPassword on FirebaseAuthentication as I'm doing with the Firebase JS SDK above? 

I was going to attempt calling `signInWithCustomToken` but `result.credential` only shows `providerId: "password"` and does not have `accessToken` available.

**Reply by robingenz** [reply]
> Is it possible that I need to sign in after calling linkWithEmailAndPassword on FirebaseAuthentication as I'm doing with the Firebase JS SDK above?

Yes, that seems to be the only difference. Here you can find our web implementation: https://github.com/capawesome-team/capacitor-firebase/blob/4ca16a6bfad418d48cf587a58be4ea3bee03adea/packages/authentication/src/web.ts#L233

**Reply by sklink** [reply]
Yeah, nothing standing out as an option there. Not sure how I would take that link and sign in using the FirebaseAuthentication package. All of them call `createSignInResult` as if part of the process should sign in the user. Maybe an edge case interaction from anonymous to first credential.

**Reply by robingenz** [reply, marked as answer]
You can simply make this one call with the Firebase JS SDK. Capacitor Firebase Authentication is based on the Firebase JS SDK on the web anyway.
